toolchain/lower/function_context.h

```diff
@@ -139,6 +146,45 @@ class FunctionContext {
   auto FinishInit(SemIR::TypeId type_id, SemIR::InstId dest_id,
                   SemIR::InstId source_id) -> void;
 
+  auto AddCallToCurrentFingerprint(SemIR::FunctionId function_id,
+                                   SemIR::SpecificId specific_id) {
+    if (!function_fingerprint_) {
+      return;
+    }
+    RawStringOstream os;
+    // TODO: Replace index with info that is translation unit independent.
+    // Using a string that includes the function_id string and the index to
+    // avoid possible collisions. This needs revisiting.
+    os << "function_id" << function_id.index;
+    current_fingerprint_.function_common_fingerprint.update(os.TakeStr());
+    // TODO: Replace index with info that is translation unit independent.
+    if (specific_id.has_value()) {
+      current_fingerprint_.function_specific_fingerprint.update(
+          specific_id.index);
+      function_fingerprint_->calls.push_back(specific_id);
+    }
+  }
+
+  auto AddTypeToCurrentFingerprint(llvm::Type* type) {
+    if (!function_fingerprint_) {
+      return;
+    }
+    if (type) {
+      RawStringOstream os;
+      type->print(os);
```

Comment:
Should there be a separator between entries here?

i.e., I assume `type->print` returns something that users can control in the type name. Suppose we have something like `type1` and `type2`, but also `type1type2`. We might want to add something like `\n` between entries, a character users *cannot* cause to be inserted, in order to guarantee these entries are unique.

Note that'd apply to all the fingerprint updates -- not sure if it's something you might want a helper for (perhaps on the fingerprinting structure), to avoid forgetting on uses.

Reply:
Added "\n" for now, added TODOs. Refactoring of "print to fingerprint" needs revisiting.